src/weights.cpp

```diff
@@ -62,25 +116,27 @@ std::vector<double> update_eff_lens(double mean, const MinCollector& tc,  const
 
   for (int i = 0; i < index.num_trans; i++) {
     if (index.target_lens_[i] < mean) {
+      // if we replace 'mean' with the proposed vector, then this case
```

Comment:
I think we can replace this with a 
means[index.target_lens[i]] > 0
or similar